src/main/java/org/littleshoot/proxy/impl/HttpProxyProtocolRequestDecoder.java

```diff
@@ -0,0 +1,52 @@
+package org.littleshoot.proxy.impl;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.util.AttributeKey;
+
+public class HttpProxyProtocolRequestDecoder extends ChannelInboundHandlerAdapter {
+
+  public static final AttributeKey<String> SOURCE_IP_ATTRIBUTE = AttributeKey.valueOf("sourceIp");
+
+  // PROXY_STRING + single space + INET_PROTOCOL + single space + CLIENT_IP + single space + PROXY_IP + single space + CLIENT_PORT + single space + PROXY_PORT + "\r\n"
+  // Example: PROXY TCP4 198.51.100.22 203.0.113.7 35646 80\r\n
+  // Source: https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/enable-proxy-protocol.html
```

Comment:
https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/enable-proxy-protocol.html

`PROXY_STRING + single space + INET_PROTOCOL + single space + CLIENT_IP + single space + PROXY_IP + single space + CLIENT_PORT + single space + PROXY_PORT + "\r\n"`

should also support IPV6

`PROXY TCP6 2001:DB8::21f:5bff:febf:ce22:8a2e 2001:DB8::12f:8baa:eafc:ce29:6b2e 35646 80\r\n`